logbook-spring-boot-autoconfigure/src/main/java/org/zalando/logbook/autoconfigure/LogbookAutoConfiguration.java

```diff
@@ -118,14 +119,25 @@ public Logbook logbook(
                 .headerFilters(headerFilters)
                 .queryFilters(queryFilters)
                 .pathFilters(pathFilters)
-                .bodyFilters(bodyFilters)
+                .bodyFilters(mergeWithTruncation(bodyFilters))
                 .requestFilters(requestFilters)
                 .responseFilters(responseFilters)
                 .strategy(strategy)
                 .sink(sink)
                 .build();
     }
 
+    private Collection<BodyFilter> mergeWithTruncation(List<BodyFilter> bodyFilters) {
+        final LogbookProperties.Write write = properties.getWrite();
+        final int maxBodySize = write.getMaxBodySize();
+        if (maxBodySize > 0) {
+            BodyFilter filter = truncate(maxBodySize);
+            // To ensure that truncation will happen after all other body filters
+            bodyFilters.add(filter);
+        }
+        return bodyFilters;
```

Comment:
1. WDYT about using new List for case if Spring provides ImmutableList for bodyFilters
2. In the documentation maxBodySize == -1 means disabled, but 0 should truncate body to empty string?
```suggestion
        if (maxBodySize < 0) {
            return bodyFilters;
        }
        // To ensure that truncation will happen after all other body filters
        List<BodyFilter> filters = new ArrayList(bodyFilters);
        filters.add(truncate(maxBodySize));
        return filters;
```

Reply:
Good suggestion! Even though tests would fail as soon as we update the version to the one with breaking change, no need to wait for it. Added in https://github.com/zalando/logbook/pull/1608/commits/bff2e1ea40919dccabca6e05ef38d618bb84e07d